src/solvers/augmented_Lagrangian_method.jl

```diff
@@ -42,8 +42,8 @@ manifold- or objective specific defaults.
 """
 mutable struct AugmentedLagrangianMethodState{
     P,
-    Pr<:AbstractManoptProblem,
-    St<:AbstractManoptSolverState,
+    Pr<:Union{F,AbstractManoptProblem} where {F},
```

Comment:
This is equivalent to just `Pr` with no upper bound.

Reply:
We can also just write F, sure. The idea here is, that it is
a) an `AbstractManoptProblem` (then `St` is a state) to indicate a(n iterative) subsolver
b) `F` (in the sense of a function, but maybe a functor) for a closed form solution, then `St` is an `AbstractEvaluationType` to indicate whether the subsolver-closed-form function allocates or works in-place.

So both these are reflected in that writing maybe for clarity – but your are right just F would also work here.

Reply:
I see, though in the b) case I think it would be more clear to store the evaluation type in a dummy `AbstractManoptSolverState` instead of making this a union.

Reply:
Well in the current form I can easily dispatch on both cases directly, but sure that change would not be too much of work (and could even be considered non-breaking since it is internal detail). However, this would affect all solvers with sub solvers, so would be a bit of work.

Reply:
I just suggest defining
```julia
struct DummyManoptSolverState{E<:AbstractEvaluationType} <: AbstractManoptSolverState end
```
(up to name) so that `sub_state` remains an actual state for consistency. Also note that `sub_state` is also used in `get_message`.

Reply:
Sounds fair.
`get_message` is mainly a summary of messages in a state (e.g. stopping criterion + stepsize and such). But yeah it might be nice to do that and sounds like a reasonable amount of work (after the current ones are finished). 

Maybe `ClosedFormSubSolverState`?

Reply:
I still feel just passing the evaluation to the keyword is nice and easy to use, but I will check what best to do there.

Reply:
Thanks! `ClosedFormSubSolverState` sounds fair.

Reply:
Wrote an issue of this to keep track.